artiq/coredevice/phaser.py

```diff
@@ -549,6 +568,50 @@ def get_dac_temperature(self) -> TInt32:
         """
         return self.dac_read(0x06, div=257) >> 8
 
+    @kernel
+    def dac_sync(self):
+        """Trigger DAC synchronisation for both output channels.
+        If the DAC-NCO is enabled, this applies NCO frequency changes."""
+        config_1f = self.dac_read(0x1f)
+        delay(.1*ms)
+        self.dac_write(0x1f, config_1f | (1 << 1))
+        self.dac_write(0x1f, config_1f & ~int32(1 << 1))
+
+    @kernel
+    def set_dac_cmix(self, fs_8_step):
+        """Set the DAC coarse mixer frequency for both channels
+
+        The selected coarse mixer frequency becomes active without explicit
+        synchronisation.
+
+        Use of the coarse mixer requires the DAC mixer to be enabled. The mixer
+        can be configured via the `dac` device_db entries.
+
+        :param fs_8_step: coarse mixer frequency shift in 125 MHz steps. This
+            should be an integer between -3 and 4 (inclusive).
+        """
+        if fs_8_step == 0:
```

Comment:
Can this become a lookup `cmix = [0x000, ...][fs_8_steps]`, or better yet just a computation with one or two bit shuffles?
Exploit the periodicity of the Nyquist zone and twos complement (`-4 === 4` and `-1 === 7` etc).

Reply:
Ack, this could be done more neatly. It's a backwards twos complement.